File: java/lib/src/main/java/com/svix/api/IngestAuthentication.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.AppPortalAccessOut;
import com.svix.models.IngestSourceConsumerPortalAccessIn;

import okhttp3.Headers;
import okhttp3.HttpUrl;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class IngestAuthentication {
    private final SvixHttpClient client;

    public IngestAuthentication(SvixHttpClient client) {
        this.client = client;
    }

    /** Get access to the Ingest Source Consumer Portal. */
    public AppPortalAccessOut consumerPortalAccess(
            final String sourceId,
            final IngestSourceConsumerPortalAccessIn ingestSourceConsumerPortalAccessIn)
            throws IOException, ApiException {
        return this.consumerPortalAccess(
                sourceId,
                ingestSourceConsumerPortalAccessIn,
                new IngestAuthenticationConsumerPortalAccessOptions());
    }

    /** Get access to the Ingest Source Consumer Portal. */
    public AppPortalAccessOut consumerPortalAccess(
            final String sourceId,
            final IngestSourceConsumerPortalAccessIn ingestSourceConsumerPortalAccessIn,
            final IngestAuthenticationConsumerPortalAccessOptions options)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/ingest/api/v1/source/%s/dashboard", sourceId));
        Map<String, String> headers = new HashMap<>();
        if (options.idempotencyKey != null) {
            headers.put("idempotency-key", options.idempotencyKey);
        }
        return this.client.executeRequest(
                "POST",
                url.build(),
                Headers.of(headers),
                ingestSourceConsumerPortalAccessIn,
                AppPortalAccessOut.class);
    }
}

File: kotlin/lib/src/main/kotlin/IngestAuthentication.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.AppPortalAccessOut
import com.svix.kotlin.models.IngestSourceConsumerPortalAccessIn
import okhttp3.Headers

data class IngestAuthenticationConsumerPortalAccessOptions(val idempotencyKey: String? = null)

class IngestAuthentication(private val client: SvixHttpClient) {
    /** Get access to the Ingest Source Consumer Portal. */
    suspend fun consumerPortalAccess(
        sourceId: String,
        ingestSourceConsumerPortalAccessIn: IngestSourceConsumerPortalAccessIn,
        options: IngestAuthenticationConsumerPortalAccessOptions =
            IngestAuthenticationConsumerPortalAccessOptions(),
    ): AppPortalAccessOut {
        val url = client.newUrlBuilder().encodedPath("/ingest/api/v1/source/$sourceId/dashboard")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }

        return client.executeRequest<IngestSourceConsumerPortalAccessIn, AppPortalAccessOut>(
            "POST",
            url.build(),
            headers = headers.build(),
            reqBody = ingestSourceConsumerPortalAccessIn,
        )
    }
}
